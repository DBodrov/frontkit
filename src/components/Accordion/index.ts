import { AccordionDetailsSuccess } from './AccordionDetailsSuccess';
import { AccordionLink } from './AccordionLink';

export const Accordion = {
    Details: AccordionDetailsSuccess,
    Link: AccordionLink,
};

export { AccordionLineType } from './AccordionDetailsSuccess';
