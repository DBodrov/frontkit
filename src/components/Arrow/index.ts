export { ArrowTypes, Arrow } from './Arrow';
