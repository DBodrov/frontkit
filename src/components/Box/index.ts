export { Box, SplitType, offsetMargin, useInsideBoxClassName } from './Box';
