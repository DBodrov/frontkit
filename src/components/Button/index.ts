export { Button, StyleTypeProp } from './Button';
export { LoadingButton } from './LoadingButton';
