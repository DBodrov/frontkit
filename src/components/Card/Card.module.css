.wrapper {
    box-shadow: 0px 4px 6px rgba(170, 187, 208, 0.16);
    border-radius: 6px;
}

.wrapper > * {
    padding: 30px;
}
.wrapper > *:not(:last-of-type) {
    border-bottom: 1px solid #eeeeee;
}
