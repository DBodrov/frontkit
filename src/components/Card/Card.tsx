import classnames from 'classnames';
import React, { ReactNode } from 'react';

import styles from './Card.module.css';

interface CardProps extends React.HTMLAttributes<HTMLElement> {
    /** Class names passed to card to change styling */
    className?: string;
    /** Inline style objects passed to card wrapper */
    style?: React.StyleHTMLAttributes<HTMLElement>;
    /** ID attribute for QA Auto-tests
     * @default Card
     * */
    dataTestId?: string;
    /** Childrens passed to card wrapper */
    children: ReactNode;
}

export function Card({ className, style, dataTestId = 'Card', children, ...rest }: CardProps): JSX.Element {
    return (
        <div data-testid={dataTestId} className={classnames(styles.wrapper, className)} style={style} {...rest}>
            {children}
        </div>
    );
}
