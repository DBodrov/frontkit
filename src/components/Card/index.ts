export { Box, SplitType } from './Box';
