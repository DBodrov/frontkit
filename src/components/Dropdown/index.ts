import { BaseDropdown } from './BaseDropdown';

export const Dropdown = BaseDropdown;
