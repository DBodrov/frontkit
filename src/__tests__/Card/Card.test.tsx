import React from 'react';
import { render } from '@testing-library/react';
import { Card } from '../../components/Card';

describe('<Card />', () => {
    const cardTestId = 'Cardddd';

    test('should card rendered', () => {
        const { getByTestId } = render(
            <Card dataTestId={cardTestId}>
                <div>1</div>
            </Card>,
        );
        const card = getByTestId(cardTestId);
        expect(card).not.toBeNull();
    });
    test('should have a childrens', () => {
        const { container } = render(
            <Card>
                <div>1</div>
                <div>2</div>
            </Card>,
        );
        expect(container).toMatchSnapshot();
    });
    test('should have a passed className', () => {
        const className = 'ItsACAAAARd';
        const { getByTestId } = render(
            <Card dataTestId={cardTestId} className={className}>
                <div>1</div>
                <div>2</div>
            </Card>,
        );
        const card = getByTestId(cardTestId);
        expect(card).toHaveClass('ItsACAAAARd');
    });
});
