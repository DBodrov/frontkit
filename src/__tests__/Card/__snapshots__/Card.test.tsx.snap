// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should have a childrens 1`] = `
<div>
  <div
    class="wrapper"
    data-testid="Card"
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
</div>
`;
